docs/sources/flow/reference/components/module.file.md

```diff
@@ -0,0 +1,165 @@
+---
+title: module.file
+labels:
+  stage: beta
+---
+
+# module.file
+
+{{< docs/shared lookup="flow/stability/beta.md" source="agent" >}}
+
+`module.file` is a *module loader* component. A module loader is a Grafana Agent Flow
+component which retreives a [module][] and runs the components defined inside of it.
+
+`module.file` simplifies the configurations for modules loaded from a file by embedding
+a [local.file][] component. This allows a single module loader to do the equivalence of
+using the more generic [module.string][] paired with a [local.file][] component.
+
+[module]: {{< relref "../../concepts/modules.md" >}}
+[local.file]: {{< relref "./local.file.md" >}}
+[module.string]: {{< relref "./module.string.md" >}}
+
+## Usage
+
+```river
+module.file "LABEL" {
+  filename  = FILENAME
+  arguments = {
+    argument1 = ARGUMENT1,
+    argument2 = ARGUMENT2,
+    ...
+  }
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`filename`       | `string`   | Path of the file on disk to watch | | yes
+`detector`       | `string`   | Which file change detector to use (fsnotify, poll) | `"fsnotify"` | no
+`poll_frequency` | `duration` | How often to poll for file changes | `"1m"` | no
+`is_secret`      | `bool`     | Marks the file as containing a [secret][] | `false` | no
+`arguments`      | `map(any)` | The values for the supported arguments in the module contents. | | no
+
+`arguments` allows us to pass parameterized input into a module. The values
+passed in `arguments` correspond to [argument blocks][] defined in the module
+source.
+
+An `argument` marked non-optional in the module being loaded is required in the
+`arguments`. It is also not valid to provide an `argument` not defined in the
+module being loaded.
+
+### File change detectors
```

Comment:
A bunch of this was copied from local.file.md but I'm not sure if we want to create a shared .md for situations or if that overcomplicates it.

Reply:
How difficult is using a shared.md? Since we only have it copied twice it doesnt feel bad to manually sync unless making the shared.md is easy.

Reply:
Not that hard, I will give it a go and we will see what it looks like.